kolibri/plugins/coach/class_summary_api.py

```diff
@@ -207,11 +209,23 @@ def data(Serializer, queryset):
     return Serializer(queryset, many=True).data
 
 
-class ClassSummaryViewSet(viewsets.ViewSet):
+class ClassSummaryPermissions(permissions.BasePermission):
+    """
+    Allow only users with admin/coach permissions on the classroom.
+    """
+
+    def has_permission(self, request, view):
+        classroom_id = view.kwargs.get('pk')
+        allowed_roles = [role_kinds.ADMIN, role_kinds.COACH]
+
+        try:
+            return request.user.has_role_for(allowed_roles, Collection.objects.get(pk=classroom_id))
+        except (Collection.DoesNotExist, ValueError):
+            return False
```

Comment:
As above, can pass `True` here, as the first line of the Viewset will give a 404.